ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/ContractKeysIT.scala

```diff
@@ -0,0 +1,554 @@
+
+    } yield {
+      assertGrpcError(
+        failedFetch,
+        LedgerApiErrors.CommandExecution.Interpreter.LookupErrors.ContractKeyNotFound,
+        Some("couldn't find key"),
+        checkDefiniteAnswerMetadata = true,
+      )
+    }
+  })
+
+  test(
+    "CKExposedByTemplate",
+    "The contract key should be exposed if the template specifies one",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    import Test.TextKey
+    val expectedKey = ledger.nextKeyId()
+    for {
+      _ <- ledger.create(party, TextKey(party, expectedKey, List.empty))
+      transactions <- ledger.flatTransactions(party)
+    } yield {
+      val contract = assertSingleton("CKExposedByTemplate", transactions.flatMap(createdEvents))
+      assertEquals(
+        "CKExposedByTemplate",
+        contract.getContractKey.getRecord.fields,
+        Seq(
+          RecordField("_1", Some(Value(Value.Sum.Party(Tag.unwrap(party))))),
+          RecordField("_2", Some(Value(Value.Sum.Text(expectedKey)))),
+        ),
+      )
+    }
+  })
+
+  test(
+    "CKExerciseByKey",
+    "Exercising by key should be possible only when the corresponding contract is available",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    import Test.TextKey
+    val keyString = ledger.nextKeyId()
+    val expectedKey = Value(
+      Value.Sum.Record(
+        Record(
+          fields = Seq(
+            RecordField("_1", Some(Value(Value.Sum.Party(Tag.unwrap(party))))),
+            RecordField("_2", Some(Value(Value.Sum.Text(keyString)))),
+          )
+        )
+      )
+    )
+    for {
+      failureBeforeCreation <- ledger
+        .exerciseByKey(
+          party,
+          TextKey.id,
+          expectedKey,
+          "TextKeyChoice",
+          Value(Value.Sum.Record(Record())),
+        )
+        .mustFail("exercising before creation")
+      _ <- ledger.create(party, TextKey(party, keyString, List.empty))
+      _ <- ledger.exerciseByKey(
+        party,
+        TextKey.id,
+        expectedKey,
+        "TextKeyChoice",
+        Value(Value.Sum.Record(Record())),
+      )
+      failureAfterConsuming <- ledger
+        .exerciseByKey(
+          party,
+          TextKey.id,
+          expectedKey,
+          "TextKeyChoice",
+          Value(Value.Sum.Record(Record())),
+        )
+        .mustFail("exercising after consuming")
+    } yield {
+      assertGrpcError(
+        failureBeforeCreation,
+        LedgerApiErrors.CommandExecution.Interpreter.LookupErrors.ContractKeyNotFound,
+        Some("dependency error: couldn't find key"),
+        checkDefiniteAnswerMetadata = true,
+      )
+      assertGrpcError(
+        failureAfterConsuming,
+        LedgerApiErrors.CommandExecution.Interpreter.LookupErrors.ContractKeyNotFound,
+        Some("dependency error: couldn't find key"),
+        checkDefiniteAnswerMetadata = true,
+      )
+    }
+  })
+
+  test(
```

Comment:
This is the only test in this file that differs from v1_8/ContractKeysTest.scala. As elsewhere, local contract keys no longer under go visibility checking, hence this test has now simplified.